pkg/ARCHITECTURE.md

```diff
@@ -0,0 +1,43 @@
+# Backend code structure
+
+Grafanas backend is written in GO using sqlite3/mysql or postgres as storage for dashboards, users etc. When Grafana was born there didnt exist much guides or direction for how to write medium sized application. So there are parts of Grafana code base that didn't quite pan out as we wanted. More about that under current rewrites! :)
+
+## Central folders of Grafanas backend
+
+| folder | description | 
+| ------- | ----------- |
+| /pkg/api | Http Handlers and routing. Almost all handler funcs are global which is something we would like to improve in the future. Handlers should be assosicated with a struct that refers to all depedencies. Ex bus |
+| /pkg/cmd | The binaries that we build. Grafana-server and grafana-cli |
```

Comment:
Just thoughts out loud - we use `cmd` for `Dispatch()` quite a bit, I wonder if it's confusing in some way

Reply:
But never in the same context. right?